x-pack/plugins/siem/public/components/toasters/index.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiGlobalToastList, Toast, EuiButton } from '@elastic/eui';
+import { noop } from 'lodash/fp';
+import React, { createContext, Dispatch, useReducer, useContext, useState } from 'react';
+import { ModalAllErrors } from './modal_all_errors';
+
+import * as i18n from './translations';
+
+export interface AppToast extends Toast {
+  errors?: string[];
+}
+
+interface ToastState {
+  toasts: AppToast[];
+}
+
+const initialToasterState: ToastState = {
+  toasts: [],
+};
+
+export type ActionToaster =
+  | { type: 'addToaster'; toast: AppToast }
+  | { type: 'deleteToaster'; id: string }
+  | { type: 'toggleWaitToShowNextToast' };
+
+export const StateToasterContext = createContext<[ToastState, Dispatch<ActionToaster>]>([
+  initialToasterState,
+  () => noop,
+]);
+
+export const useStateToaster = () => useContext(StateToasterContext);
+
+interface ManageGlobalToasterProps {
+  children: React.ReactNode;
+}
+
+export const ManageGlobalToaster = ({ children }: ManageGlobalToasterProps) => {
+  const reducerToaster = (state: ToastState, action: ActionToaster) => {
+    switch (action.type) {
+      case 'addToaster':
+        return { ...state, toasts: [...state.toasts, action.toast] };
+      case 'deleteToaster':
+        return { ...state, toasts: state.toasts.filter(msg => msg.id !== action.id) };
+      default:
+        return state;
+    }
+  };
+
+  return (
+    <StateToasterContext.Provider value={useReducer(reducerToaster, initialToasterState)}>
+      {children}
+    </StateToasterContext.Provider>
+  );
+};
+
+interface GlobalToasterProps {
+  toastLifeTimeMs?: number;
+}
+
+export const GlobalToaster = ({ toastLifeTimeMs = 5000 }: GlobalToasterProps) => {
+  const [{ toasts }, dispatch] = useStateToaster();
+  const [isShowing, setIsShowing] = useState(false);
+  const [toastInModal, setToastInModal] = useState<AppToast | null>(null);
+
+  const toggle = (toast: AppToast) => {
+    if (isShowing) {
+      dispatch({ type: 'deleteToaster', id: toast.id });
+      setToastInModal(null);
+    } else {
+      setToastInModal(toast);
+    }
+    setIsShowing(!isShowing);
+  };
+
+  return (
+    <>
+      {toasts != null && toasts.length > 0 && !isShowing && (
```

Comment:
I don't think you need the nullcheck as `useStateToaster()`'s  `useContext()`'s default value (ToastState) can't be null:

``` ts
interface ToastState {
  toasts: AppToast[];
}
```